proposals/0011-inline-spirv.md

```diff
@@ -178,15 +185,15 @@ SPIRV.html#OpTypeArray) takes an id for the element type and an id for the
 element length, so an array of 16 integers could be declared as
 
 ```
-vk::SpirvOpaqueType</* OpTypeArray */ 28, int, 16>
+vk::SpirvOpaqueType</* OpTypeArray */ 28, int, vk::integral_constant<uint, 16> >
```

Comment:
If I understand correctly, the goal of adding this type is to know upfront the destination type of the constant without having to guess?
But why was `vk::ext_literal` bad? What issues did you find?

Reply:
iirc the specific critique from @llvm-beanz was that it was defined as a function that returned `void`, and we were "looking through" the function call in the AST to get the value. We could fix it by making a function return a value of, say, `vk::Literal<T>` type, but Chris suggested making it a type qualifier, which is what I've done here.

I do wonder if maybe this is a very specific thing to add a new type qualifier for – it's only relevant for the `SpirvType` template.

Reply:
Ahh right, I see. Thanks you both for the explanation!
Got it now.